scm-plugin/src/main/resources/archetype-resources/src/main/java/HelloWorldSCMSource.java

```diff
@@ -0,0 +1,64 @@
+package $package;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.TaskListener;
+import hudson.scm.SCM;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMHeadEvent;
+import jenkins.scm.api.SCMHeadObserver;
+import jenkins.scm.api.SCMRevision;
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.SCMSourceCriteria;
+import jenkins.scm.api.SCMSourceDescriptor;
+import org.jenkinsci.Symbol;
+
+import java.io.IOException;
+import java.util.Set;
+
+public class HelloWorldSCMSource extends SCMSource {
+    @Override
+    protected void retrieve(SCMSourceCriteria scmSourceCriteria,
+                            @NonNull SCMHeadObserver scmHeadObserver,
+                            SCMHeadEvent<?> scmHeadEvent,
+                            @NonNull TaskListener taskListener) throws IOException, InterruptedException {
+        // TODO
+    }
+
+    @Override
+    public SCMRevision getTrustedRevision(SCMRevision revision,
+                                          TaskListener listener) throws IOException,
+            InterruptedException {
+        return super.getTrustedRevision(revision, listener); // TODO
+    }
+
+
+    @Override
+    protected Set<String> retrieveRevisions(TaskListener listener) throws IOException, InterruptedException {
+        return super.retrieveRevisions(listener); // TODO
+    }
+
+    @Override
+    protected SCMRevision retrieve(String thingName,
+                                   TaskListener listener) throws IOException, InterruptedException {
+        return super.retrieve(thingName, listener); // TODO
```

Comment:
Just overriding this will improperly make this SCM [appear under the **Modern** list](https://github.com/jenkinsci/workflow-cps-global-lib-plugin/blob/64bbefaede3e1d5dca83a02aa66befd8263c08a2/src/main/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetriever.java#L155), which we do not want.